### ISSUE TITLE
Is it possible to postpone the access_token expiration with every call?

### ISSUE THREAD

**Issue by carlo161** [greenpau/caddy-security#335]
When I successfully login and gets forwarded to my webpage I obtain the following:

![2024-04-17_12h16_50](https://github.com/greenpau/caddy-security/assets/30337724/3047365b-26ff-4697-9e48-539ce8675fb1)

I would like to implement some sort of a timeout: if you do not interact with the page in 10 minutes, at the next call you will have to login again.
Right now I am using:

```
    authentication portal my_portal {
      crypto default token lifetime 600
      cookie lifetime 7200
``` 

But after 20 minutes I have to login again.
With every call the expiration of the session cookie gets extended but the access_token one no.
Is it possible to do that?

**Comment by jspadaro** [comment]
The issue appears to me that that token is a jwt that is hardcoded to expire in 900 seconds (15 minutes)

It's pretty annoying with SSO and really annoying if you do local logins or are streaming anything for any duration.  

I'm actually not clear why you can even set a cookie lifetime if this limitation is still in place.

I may try to patch something myself for my own use, but I would really love to see something similar as an official change.

**Comment by carlo161** [comment]
> crypto default token lifetime 7200 cookie lifetime 14440
> 
> this would be 2 hours expiration

I am not trying to enlarge the lifetime, but to reset it with every interaction. If the user keeps on working with the page the lifetime is potentially eternal, but as soon as he stops, after for example 10 minutes the token expires.

**Comment by greenpau** [comment]
@carlo161 , are you asking for the ability to refresh the token just before it expires?

**Comment by greenpau** [comment]
As long as the user is active?

**Comment by jspadaro** [comment]
Ah, I was mistaken.  Either way, my overarching confusion was "Why have a longer cookie lifetime if the JWT expiring kills the session?"

Anyway, back to this issue - yes, @greenpau , I think the ask is to make the token auto-refresh / have the expiration extend to now+increment on every request

**Comment by carlo161** [comment]
Yes, right. I would like to refresh the token with every call, as long as the user is active.

**Comment by alyssondiasmec** [comment]
Following this for further information. Since i would like to refresh the token on every call too.

**Comment by alyssondiasmec** [comment]
Any update on this? This plugin already support token refresh?

**Comment by amc1999** [comment]
It is a critical issue for our project: users have to stay logged in for as long as they are active (GET, POST,....). The only option for me now is to set lifetime = 2 years, which is not a proper solution but rather a desperate measures.

**Comment by wisammechano** [comment]
this plugin is very nice but it does require some refreshing mechanism. I am working with logs and suddenly it fails to fetch and I have to refresh and log in again which is quite disruptive.

**Comment by adg-mh** [comment]
I'm also interested in this. @greenpau is this on the table or not something you'd want to add?
